xformers/components/multi_head_dispatch.py

```diff
@@ -143,6 +143,12 @@ def forward(
         self._check(value, "value")
         self._check(key, "key")
 
+        if query.size(0) < key.size(0):
+            query = query.expand_as(key)
+        elif query.size(0) > key.size(0):
+            key = key.expand_as(query)
+            value = value.expand_as(query)
```

Comment:
I don't think this is the best way to do this. Currently it will require the `Q_SEQLEN` and `K_SEQLEN` to match, which is strictly not a requirement... Any thoughts, @blefaudeux?

Reply:
It also assumes that `key.size() == value.size()` which is probably OK because we already had that assumption, I think.

Reply:
hmm I would need to check how the expand_as() works, technically for the computations you don't need Q and K to have the same sequence length so we should not change that. how about 
`
batch = max([q.shape[0], k.shape[0], v.shape[0]])
q , k, v = map(lambda x: x.expand(batch, -1, -1), [q, k v])
`
? top of head, this could be wrong but at least it's not long (who cares about correctness ? :D)

Reply:
Obvious in hindsight. 😄